### DISCUSSION TITLE
Distance between an O-D pair using the road network

### DISCUSSION THREAD

**Discussion by wesleneuy** [ropensci/stplanr#484] [General]
Hi all! I'm fairly new to stplanr and I hope to use it to calculate the vehicle km/distance traveled. I'm wondering if it's possible to get the distance between an origin and destination TAZ using a road network (extracted from OSM) in stplanr.  I've only used stplanr to get the euclidean distance from an O-D pair. If anyone could share a code vignette or any other resource, that would be very helpful!

For reference, my data set has the following info:

Household #
Trip #
Trip Total
Origin TAZ Code
Departure time at origin
Travel mode at Departure
TAZ Code of 1st Transfer Point
Departure time at origin
Travel mode at Departure
TAZ Code of 2nd Transfer Point
Departure time at origin
Travel mode at Departure
TAZ Code of 3rd Transfer Point
Departure time at origin
Travel mode at Departure
TAZ Code of 4th Transfer Point
Departure time at origin
Travel mode at Departure
Destination TAZ Code
Arrival Time at Destination
Type of Destination
Trip Purpose

**Comment by Robinlovelace** [comment]
Hi @wesleneuy if you can provide a reproducible example that will help demonstrate exactly how to solve the problem, but here's some links to code/descriptions that may help.

Slope sensitive routing idea: https://github.com/sfnetworks/sloperouting

'Route local' idea (note, planning to switch this to using {sfnetworks} as described here #483 ): https://docs.ropensci.org/stplanr/reference/route.html

@agila5 and others may have more examples. If you provide some example input data we can provide some reproducible code.
